engine/src/main/java/org/hibernate/search/spi/impl/IndexedTypesSets.java

```diff
@@ -81,4 +81,67 @@ else if ( types.length == 1 ) {
 		}
 	}
 
+	public static IndexedTypesSet composite(final IndexedTypesSet set, final IndexedTypeIdentifier additionalId) {
+		if ( set.isEmpty() ) {
+			return additionalId.asTypeSet();
+		}
+		//We have only one internal representation; casting as I don't want
+		//to expose additional methods on the IndexedTypesSet interface.
+		HashSetIndexedTypesSet casted = (HashSetIndexedTypesSet) set;
+		if ( casted.contains( additionalId ) ) {
+			return set;
+		}
+		else {
+			HashSet<IndexedTypeIdentifier> newSet = new HashSet<>( set.size() + 1 );
+			for ( IndexedTypeIdentifier existingId : set ) {
+				newSet.add( existingId );
+			}
+			newSet.add( additionalId );
+			return new HashSetIndexedTypesSet( newSet );
+		}
+	}
+
+	public static IndexedTypesSet composite(final IndexedTypesSet setA, final IndexedTypesSet setB) {
```

Comment:
`IndexedTypesSet` is an SPI exposed type, while IndexedTypesSets helper is currently not.

I guess we might end up realising that we need to expose such helpers too, but I'd hope to try avoiding that, and then when we have more stuff cleaned up see if it's really useful to integrators to access such helpers.